zilliqa/src/state.rs

```diff
@@ -351,7 +389,9 @@ pub mod contract_addr {
     pub const INTERSHARD_BRIDGE: Address = Address::new(*b"\0\0\0\0\0\0\0\0ZQINTERSHARD");
     /// Address of the shard registry - only present on the root shard.
     pub const SHARD_REGISTRY: Address = Address::new(*b"\0\0\0\0\0\0\0\0\0\0\0\0\0ZQSHARD");
-    pub const DEPOSIT: Address = Address::new(*b"\0\0\0\0\0\0\0\0\0\0ZILDEPOSIT");
+    /// Address of EIP 1967 proxy for Deposit contract
+    pub const DEPOSIT_PROXY: Address = Address::new(*b"\0\0\0\0\0ZILDEPOSITPROXY");
+    pub const DEPOSIT_PROXY_OWNER: Address = Address::new(*b"ZILDEPOSITPROXYOWNER");
```

Comment:
As mentioned above, can't we use the zero address instead which can't be impersonated. neither by EOAs nor by other contracts to maliciously upgrade the deposit contract?